### ISSUE TITLE
Failure to start: ValidateConfig: error normalizing listen addrs

### ISSUE THREAD

**Issue by rkfg** [lightningnetwork/lnd#6339]
### Background

Lnd fails to start after 7dfe4018ce2f95c2f6a9ff04e370f668fc755138 has been merged. There are no messages in the log right after the start but after 2 minutes this is returned:
```
[WRN] LTND: Error validating config: ValidateConfig: error normalizing REST listen addrs: parse address 0.0.0.0:8081 failed: EOF
failed to load config: ValidateConfig: error normalizing REST listen addrs: parse address 0.0.0.0:8081 failed: EOF
```

### Your environment

* version of `lnd` 7dfe4018ce2f95c2f6a9ff04e370f668fc755138
* which operating system (`uname -a` on *Nix) Debian 11.2 amd64 (and also Raspbian arm64)

I bisected `lnd` and found that this is the bad commit. My config has been working fine for half a year. Maybe there's something expected but missing in my `lnd.conf` but it shouldn't freeze for 2 minutes and then drop this hard to understand error. On my main node (on Raspbian/amd64) the error is slightly different:
```
[WRN] LTND: Error validating config: ValidateConfig: error normalizing p2p listen addrs: parse address 0.0.0.0:9735 failed: EOF
failed to load config: ValidateConfig: error normalizing p2p listen addrs: parse address 0.0.0.0:9735 failed: EOF
```
But otherwise the behavior is the same.

**Comment by Roasbeef** [comment]
> . My config has been working fine for half a year. Maybe there's something expected but missing in my lnd.conf but it shouldn't freeze for 2 minutes and then drop this hard to understand error.

Thanks for reporting this! If you're running the master branch, then you may occasionally run into things like this, with the expectation that by the final release (aided by issue reports like this) these type of issues will have been caught. 

That commit is interesting, as it's mainly updating to use new Bitcoin libraries, and this looks to be a network address parsing issue: https://github.com/lightningnetwork/lnd/blob/master/lncfg/address.go#L34-L41

Perhaps it's actually some other commit instead?

**Comment by Roasbeef** [comment]
Has anything else changed in your set up? Are you running a tor node?

**Comment by rkfg** [comment]
I ran a regular `git bisect` using `master` as `bad` and 72aec1722b4a04b3df144b3363ecb1c59e5078a8 as `good` (the last commit I built lnd from before). It eventually pointed at 7dfe4018ce2f95c2f6a9ff04e370f668fc755138. It might be a dependency issue as well.

I haven't changed anything really, I even tried it on two different nodes. One runs on RPi 4/Raspbian/arm64 mainnet, the other runs on Debian stable/amd64 testnet. Both show the same issue.

Tor seems to be the problem though! After I disabled it `lnd` managed to start normally (and after enabling it back the issue is also back). At least on one of the nodes (testnet), that on RPi (mainnet) takes a long while to launch so I prefer to keep it running.

**Comment by Roasbeef** [comment]
Are you running with the Tor health check active? Without it, if tor restarts then lnd's connection won't be properly re-established. Here's the config section: https://github.com/lightningnetwork/lnd/blob/master/sample-lnd.conf#L1000

**Comment by rkfg** [comment]
The whole heathcheck section is commented. I don't restart Tor though.

**Comment by Roasbeef** [comment]
Gotcha, I'd recommend turning on the Tor one, as it'll also detect any unexpected disruption of the control port connection and re-create any relevant onion services as well. Or if for w/e reason the tor daemon restarts, then this makes sure lnd either also restarts or can heal things. Otherwise, you'll need to manually restart your lnd node each time something weird happens with the tor daemon.

**Comment by rkfg** [comment]
Okay, but the issue isn't solved. I added the healthcheck lines from the sample, enabled Tor and `lnd` doesn't start again with the same error. I don't see how this is fixed because if I build lnd from the previous commit it runs just fine on the same node with zero changes to the config or Tor. Also the error is very confusing. Also, I don't think this behavior is good for the default config with these lines commented.

**Comment by Roasbeef** [comment]
> Tor seems to be the problem though! After I disabled it lnd managed to start normally (and after enabling it back the issue is also back). At least on one of the nodes (testnet), that on RPi (mainnet) takes a long while to launch so I prefer to keep it running.

Missed this the first time around: so the issue is you can't start w/ tor at all w/ that set up?

**Comment by rkfg** [comment]
> so the issue is you can't start w/ tor at all w/ that set up?

Exactly, if it's enabled `lnd` freezes for exactly 2 minutes, then errors out with the message I posted in the OP. If Tor is disabled it starts normally.

**Comment by Roasbeef** [comment]
Still don't see how those commits are related (could be something with bisect not being aware of merge commits the way it's being used?), but I think this section needs to be updated to allow the "all interfaces" IP to pass through to normal TCP resolution: https://github.com/lightningnetwork/lnd/blob/262591c3331c2170697fe89b8c49ab857db0adf7/lncfg/address.go#L235

**Comment by rkfg** [comment]
Well, no idea but 8ee9fc837b1a4a575cb37b413ae11ee97dd397fe is working and this 7dfe4018ce2f95c2f6a9ff04e370f668fc755138 (which is the next one) is not.

**Comment by Roasbeef** [comment]
Current hunch is that we fixed a bug on the btcd/btcsuite side somewhere, that unmasked a bug here where we attempt to use Tor DNS resolution for the "all interfaces" IP. Likely need to ignore that here: https://github.com/lightningnetwork/lnd/blob/262591c3331c2170697fe89b8c49ab857db0adf7/lncfg/address.go#L235
